File: core/src/main/java/compf/core/engine/pokemon/EVDistribution.java
package compf.core.engine.pokemon;


import static compf.core.engine.pokemon.Constants.H;
class Constants
{
    public static final int  H=252;

}

public enum  EVDistribution {
    HP_ATT(new int[]{H, H, 0, 0, 0, 0}),
    HP_DEF(new int[]{H, 0, H, 0, 0, 0}),
    HP_SATT(new int[]{H, 0, 0, H, 0, 0}),
    HP_SDEF( new int[]{H, 0, 0, 0, H, 0}),
    HP_SPEED(new int[]{H, 0, 0, 0, 0, H}),

    ATT_DEF(new int[]{0, H, H, 0, 0, 0}),
    ATT_SATT(new int[]{0, H, 0, H, 0, 0}),
    ATT_SDEF(new int[]{0, H, 0, 0, H, 0}),
    ATT_SPEED( new int[]{0, H, 0, 0, 0, H}),


    DEF_SATT(new int[]{0, 0, H, H, 0, 0}),
    DEF_SDEF( new int[]{0, 0, H, 0, H, 0}),
    DEF_SPEED(new int[]{0, 0, H, 0, 0, H}),


    SATT_SDEF(new int[]{0, 0, 0, H, H, 0}),
    SATT_SPEED(new int[]{0, 0, 0, H, 0, H}),

    SDEF_SPEED(new int[]{0, 0, 0, 0, H, H}),
    ;
    EVDistribution(int [] evs){
    this.evs=evs;
    }

    public int[] evs;

}

File: teamBuilder/src/main/kotlin/pokeclass/EVDistribution.kt
package pokeclass

import teambuilder.ThreatFinder
public const val  H=252
enum class EVDistribution {
    HP_ATT(intArrayOf(H,H, 0, 0, 0, 0)),
    HP_DEF(intArrayOf(H, 0, H, 0, 0, 0)),
    HP_SATT(intArrayOf(H, 0, 0, H, 0, 0)),
    HP_SDEF( intArrayOf(H, 0, 0, 0, H, 0)),
    HP_SPEED(intArrayOf(H, 0, 0, 0, 0, H)),

    ATT_DEF(intArrayOf(0, H, H, 0, 0, 0)),
    ATT_SATT(intArrayOf(0, H, 0, H, 0, 0)),
    ATT_SDEF(intArrayOf(0, H, 0, 0, H, 0)),
    ATT_SPEED( intArrayOf(0, H, 0, 0, 0, H)),


    DEF_SATT(intArrayOf(0, 0, H, H, 0, 0)),
    DEF_SDEF( intArrayOf(0, 0, H, 0, H, 0)),
    DEF_SPEED(intArrayOf(0, 0, H, 0, 0, H)),


    SATT_SDEF(intArrayOf(0, 0, 0, H, H, 0)),
    SATT_SPEED(intArrayOf(0, 0, 0, H, 0, H)),

    SDEF_SPEED(intArrayOf(0, 0, 0, 0, H, H)),
    ;
    constructor(evs:IntArray){
    this.evs=evs
    }

    val evs:IntArray
}